pytensor/link/c/cmodule.py

```diff
@@ -466,13 +467,13 @@ def get_module_hash(src_code, key):
         if isinstance(key_element, tuple):
             # This should be the C++ compilation command line parameters or the
             # libraries to link against.
-            to_hash += list(key_element)
+            to_hash += [str(e) if isinstance(e, Path) else e for e in key_element]
```

Comment:
Code below expects this to be all-string, and calling `str(.)` on a string doesn't matter
```suggestion
            to_hash += [str(e) for e in key_element]
```